src/detect-engine.c

```diff
@@ -1161,6 +1161,47 @@ void InspectionBufferCopy(InspectionBuffer *buffer, uint8_t *buf, uint32_t buf_l
     }
 }
 
+/** \brief Check content string compatibility with transforms
+ *
+ *  The "checkstr" is presented to the transforms so that each transform may
+ *  validate that it's compatible with the transform.
+ *
+ *  When a transform indicates the string is incompatible, none of the subsequent
+ *  transforms, if any, are checked.
+ *
+ *  \param de_ctx Detection engine context.
+ *  \param sm_list The SM list id.
+ *  \param checkstr the content string being validated
+ *  \param namestr returns the name of the transform that is incompatible with checkstr.
+ *
+ *  \retval true (false) If any of the transforms indicate the string is not
+ *  compatible.
+ **/
+bool DetectBufferTypeValidateTransform(DetectEngineCtx *de_ctx, int sm_list,
+        const char *checkstr, const char **namestr)
```

Comment:
following the comments above of what to check, the `checkstr` arg here would need to be changed to a `const uint8_t *` combined with a length param.